src/cache.rs

```diff
@@ -44,7 +44,7 @@ impl<Fs: FileSystem> Cache<Fs> {
     pub fn value(&self, path: &Path) -> CachedPath {
         let hash = {
             let mut hasher = FxHasher::default();
-            path.as_os_str().hash(&mut hasher);
+            path.hash(&mut hasher);
```

Comment:
```suggestion
            #[cfg(windows)]
            path.hash(&mut hasher);
            #[cfg(not(windows))]
            path.as_os_str().hash(&mut hasher);
```
It might be possible to do this. But I'm not sure that's safe for all non-Windows OSs.